### PR TITLE
store: loop over smaller list

### PR DESCRIPTION
**Pull request by kimmylin**
- avoid N square loop
- loop over smaller dataset and test membership for the bigger set

### CODE REVIEW

File: store/subscription.go [hunk 1]
```diff
@@ -213,63 +213,68 @@ func (s *subscription) insert(ctx context.Context, events []*model.Transfer) (er
 	}
 
 	// Calculate tx fee
-	fees := make(map[gethCommon.Address]*big.Int)
+	fees := make(map[string]*big.Int)
```
Comment (markya0616) [inline store/subscription.go]: Suggest to use `common.Address` as key.

File: store/subscription.go [hunk 2]
```diff
@@ -298,10 +303,9 @@ func (s *subscription) insert(ctx context.Context, events []*model.Transfer) (er
...
 	// Add diff in total balances
-	for _, sub := range subs {
-		addr := gethCommon.BytesToAddress(sub.Address)
-		for token, addrs := range addrDiff {
-			d, ok := addrs[addr]
+	for token, addrs := range addrDiff {
+		for addr, d := range addrs {
+			sub, ok := allSubs[addr]
 			if !ok {
 				continue
```
Comment (markya0616) [inline store/subscription.go]: Print a warming here. It should not be happen.

File: store/subscription.go (hunk 1 above)
Comment (kimmylin) [reply store/subscription.go]: Q: i didn't understand why we want to convert a 32 byte tx hash to 20 byte address?

Comment (markya0616) [reply store/subscription.go]: Ha, it's a bug and should convert to `common.Hash`.

Comment (kimmylin) [reply store/subscription.go]: ha. thanks! i was confused. fixed.

File: store/subscription.go (hunk 2 above)
Comment (kimmylin) [reply store/subscription.go]: done

**Pull request merged**